app/controllers/shopify_app/callback_controller.rb

```diff
@@ -26,6 +31,29 @@ def callback
 
     private
 
+    def jwt_callback
+      if valid_jwt_auth?
+        create_user_session_from_jwt_callback
+        head(:ok)
+      else
+        head(:unauthorized)
+      end
+    end
+
+    def valid_jwt_auth?
+      auth_hash && jwt_shopify_domain == shop_name && jwt_shopify_user_id == associated_user_id
```

Comment:
It doesn't look like we log anywhere else in the old callback code, but I still wonder if our future selves will be happier if we log why the jwt was invalid (or really, why the callback failed)

Reply:
Both the JWT middleware and the ominauth middleware have pretty decent logging, so I think we can rely on those two sources to give enough debugging information (and supplement later on when/if we discover particular cases that aren't covered well).